src/modules/schnorr_adaptor/tests_impl.h

```diff
@@ -0,0 +1,957 @@
+
+        CHECK(secp256k1_schnorr_adaptor_extract(CTX, t, sig[sig_idx], msg[sig_idx], &pk));
+        CHECK(secp256k1_memcmp_var(t, adaptor[sig_idx], 33) == 0);
+    }
+
+    /* Test overflowing s */
+    CHECK(secp256k1_schnorr_adaptor_presign(CTX, sig[0], msg[0], &keypair, adaptor[0], NULL) == 1);
+    CHECK(secp256k1_schnorr_adaptor_extract(CTX, t, sig[0], msg[0], &pk));
+    CHECK(secp256k1_memcmp_var(t, adaptor[0], 33) == 0);
+    memset(&sig[0][33], 0xFF, 32);
+    CHECK(!secp256k1_schnorr_adaptor_extract(CTX, t, sig[0], msg[0], &pk));
+    CHECK(secp256k1_memcmp_var(t, adaptor[0], 33) != 0);
+
+    /* Test negative s */
+    CHECK(secp256k1_schnorr_adaptor_presign(CTX, sig[0], msg[0], &keypair, adaptor[0], NULL) == 1);
+    CHECK(secp256k1_schnorr_adaptor_extract(CTX, t, sig[0], msg[0], &pk));
+    CHECK(secp256k1_memcmp_var(t, adaptor[0], 33) == 0);
+    secp256k1_scalar_set_b32(&s, &sig[0][33], NULL);
+    secp256k1_scalar_negate(&s, &s);
+    secp256k1_scalar_get_b32(&sig[0][33], &s);
+    CHECK(secp256k1_schnorr_adaptor_extract(CTX, t, sig[0], msg[0], &pk));
+    CHECK(secp256k1_memcmp_var(t, adaptor[0], 33) != 0);
+}
+#undef N_SIGS
+
+void test_schnorr_adaptor_adapt_extract_sec(void) {
+    unsigned char sk[32];
+    secp256k1_xonly_pubkey pk;
+    secp256k1_keypair keypair;
+    secp256k1_scalar adaptor_scalar;
+    secp256k1_gej tj;
+    secp256k1_ge tg;
+    unsigned char msg[32];
+    unsigned char sig[65];
+    unsigned char sig64[64];
+    unsigned char secadaptor[32];
+    unsigned char aux_rand[32];
+    unsigned char adaptor[33];
+    unsigned char t[33];
+    unsigned char t2[32];
+    size_t size = 33;
+
+    secp256k1_testrand256(sk);
+    secp256k1_testrand256(msg);
+    secp256k1_testrand256(secadaptor);
+    secp256k1_testrand256(aux_rand);
+    secp256k1_scalar_set_b32(&adaptor_scalar, secadaptor, NULL);
+    secp256k1_ecmult_gen(&CTX->ecmult_gen_ctx, &tj, &adaptor_scalar);
+    secp256k1_ge_set_gej(&tg, &tj);
+    CHECK(secp256k1_eckey_pubkey_serialize(&tg, adaptor, &size, 1) == 1);
+    CHECK(secp256k1_keypair_create(CTX, &keypair, sk) == 1);
+    CHECK(secp256k1_keypair_xonly_pub(CTX, &pk, NULL, &keypair) == 1);
+    CHECK(secp256k1_schnorr_adaptor_presign(CTX, sig, msg, &keypair, adaptor, NULL) == 1);
+    CHECK(secp256k1_schnorr_adaptor_extract(CTX, t, sig, msg, &pk));
+    CHECK(secp256k1_memcmp_var(t, adaptor, 33) == 0);
+    CHECK(secp256k1_schnorr_adaptor_adapt(CTX, sig64, sig, secadaptor) == 1);
+    CHECK(secp256k1_schnorr_adaptor_extract_sec(CTX, t2, sig, sig64) == 1);
+    CHECK(secp256k1_memcmp_var(t2, secadaptor, 32) == 0);
+}
+
+void run_schnorr_adaptor_tests(void) {
+    int i;
+    run_adaptor_nonce_function_bip340_tests();
+
+    test_schnorr_adaptor_api();
+    test_schnorrsig_sha256_tagged();
+    test_schnorr_adaptor_vectors();
+    for (i = 0; i < COUNT; i++) {
+        test_schnorr_adaptor_presign();
+        test_schnorr_adaptor_extract();
+        test_schnorr_adaptor_adapt_extract_sec();
```

Comment:
I don't quite understand the intentions of these tests. `presign` seems to be tested in both `test_schnorr_adaptor_extract()` and `test_schnorr_adaptor_adapt_extract_sec` so `test_schnorr_adaptor_presign` seems to be redundant. I think it would be sufficient to have one "correctness" test that runs the full protocol (essentially `test_schnorr_adaptor_adapt_extract_sec`) and one test for failures due to bitflips or negating `s` (essentially `test_schnorr_adaptor_extract`).

Reply:
The `presign` test could benefit from testing with [different nonce functions](https://github.com/BlockstreamResearch/secp256k1-zkp/blob/master/src/modules/ecdsa_adaptor/tests_impl.h#L894) (failing, returns 0, overflowing).

Reply:
_side note:_ The ECDSA adaptor module performs these tests under a single [`adaptor_tests` function](https://github.com/BlockstreamResearch/secp256k1-zkp/blob/master/src/modules/ecdsa_adaptor/tests_impl.h#L868).